teckyl/tc/lang/lexer.cc

```diff
@@ -17,6 +17,8 @@
 
 #include <cstring>
 
+#include <llvm/ADT/Twine.h>
+#include <llvm/Support/ErrorHandling.h>
```

Comment:
I would prefer preserving the possible use of the lexer / parser / semantic analysis as library functions and see all exceptions preserved / wrapped in a preprocessor macro.